yt_dlp/extractor/tvplay.py

```diff
@@ -30,162 +29,12 @@ class TVPlayIE(InfoExtractor):
                         mtg:|
                         https?://
                             (?:www\.)?
-                            (?:
-                                tvplay(?:\.skaties)?\.lv(?:/parraides)?|
-                                (?:tv3play|play\.tv3)\.lt(?:/programos)?|
-                                tv3play(?:\.tv3)?\.ee/sisu|
-                                (?:tv(?:3|6|8|10)play)\.se/program|
-                                (?:(?:tv3play|viasat4play|tv6play)\.no|(?:tv3play)\.dk)/programmer|
-                                play\.nova(?:tv)?\.bg/programi
-                            )
+                            (?:play\.nova(?:tv)?\.bg/programi)
                             /(?:[^/]+/)+
                         )
                         (?P<id>\d+)
                     '''
     _TESTS = [
-        {
```

Comment:
Dont remove the tests, just change them to their new urls

Reply:
I don't think I understand which tests you are referring to?

Reply:
The removed URLs still work and are just redirected to the new site. So you should replace the tests with the redirected URLs. Eg: `http://www.tvplay.lv/parraides/vinas-melo-labak/418113` redirects to `https://tv3play.skaties.lv/series/vinas-melo-labak,serial-1024464`. So the first test in `TVPlayIE` should be changed to that URL

Reply:
Well, true. It redirects but to an invalid URL. It cannot work with the tvplay extractor since MTG no longer owns that service. If you try to use the ID there in the redirected URL in the tvplay extractor you get nothing. It also won't work in the tvplayhome extractor since it doesn't redirect to an episode like the original URL before the redirect. It just redirects to a season summary page. Are such pages even supposed to be supported? Or should it just support direct episode pages so we know which one to play?

Reply:
If we are supposed to support season pages how would we do that? Follow the link shown to the episode page and then do the extraction of the episode information or try to query the API for the latest season information and then pick the latest episode?

Reply:
Basically make a playlist info extractor like `TVPlaySeasonIE`, haven't looked the site specifically but if you can grab a list of episode URLs you should be able to simply have it return a bunch of `url_transparent`s and have the main extractor handle it.

Reply:
Since there is a video in the final page, I assumed that is the same video as the original URL. But now I see it is not (and is the first video of the series?)

Yes, such pages can be supported, but that would be a different extractor. You were right to remove the tests

Reply:
Another issue though, is that those URLs are being captured by this extractor. This should not happen

Reply:
I might be confused here but it seems like we seem to assume that tvplay is one service with identifiers working across multiple APIs and multiple different businesses. At one time. Around 4-5 years ago MTG owned all of these so one tvplay extractor worked fine. It was one API. 

Then they restructured and sold a lot of what used to be MTG services over the years. The URLs I removed were all URLs with identifiers linked to the MTG API which is no more. Some seem to redirect, but not to the actual episodes that were the original URL when it was owned by MTG.

What we have left is the Viafree and the Tvplayhome extractors and someone now also asked for play.tv2.no in this pull request. None of these have anything in common at all except that their URLs seem similar. All three of these are owned by different companies having different APIs and identifiers. So, this entire concept of tvplay as one single concept working across multiple services/businesses/APIs and regions to me is very strange to keep.

Am I being unreasonable?

Reply:
>  Since there is a video in the final page, I **assumed** that is the same video as the original URL. But **now I see it is not**
> Yes, such pages can be supported, but that would be a different extractor. **You were right to remove the tests**

No. You are right. I had not understood the situation at first. See my above comments

Reply:
> Another issue though, is that those URLs are being captured by this extractor. This should not happen
> 
> ```
> ➤ yt-dlp https://tv3play.skaties.lv/series/vinas-melo-labak,serial-1024464
> [TVPlayHome] 1024464: Downloading JSON metadata
> ERROR: [TVPlayHome] 1024464: 1024464: Failed to parse JSON  (caused by JSONDecodeError('Expecting value: line 1 column 1 (char 0)')); please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
> ```

Sure. It actually had the same issue before they updated the URLs but of course I can fix that.

Reply:
> Basically make a playlist info extractor like `TVPlaySeasonIE`, haven't looked the site specifically but if you can grab a list of episode URLs you should be able to simply have it return a bunch of `url_transparent`s and have the main extractor handle it.

Is this possible? A playlist extractor? Cool. Do you have some example code from another extractor I could look at? I like the idea and if I could implement something like that here I can use it when I implement other extractors. I mean I can totally see the nice use case to just grab a season page and just batch the season.

Though I do guess if it adds a lot of complexity it might not be worth the effort compared to just making it a little more elegant when a URL is not a supported episode/live/clip URL.

Reply:
Just search for IEs ending with `Playlist`, `Show`, `Series`. There are plenty of examples. `Zee5SeriesIE` in zee5.py is a simple example

If u r doing it, keep it to a separate PR though